components/Seo.php

```diff
@@ -58,4 +36,152 @@ public function defineProperties()
+        }
+
+        $settings = Settings::instance();
+
+        if ($settings->site_name_position == 'prefix') {
+            $title = "{$settings->site_name} {$settings->site_name_separator} {$title}";
+        } else if ($settings->site_name_position == 'suffix') {
+            $title = "{$title} {$settings->site_name_separator} {$settings->site_name}";
+        }
+
+        return $title;
+    }
+
+    public function getDescription()
+    {
+        $description = Settings::instance()->description;
+        if (isset($this->page['description'])) {
+            $description = $this->page['description'];
+        }
+
+        if (isset($this->viewBagProperties['meta_description'])) {
+            $description = $this->viewBagProperties['meta_description'];
+        }
+
+        if (isset($this->viewBagProperties['localeMeta_description[' . $this->locale . ']'])) {
+            $description = $this->viewBagProperties['localeMeta_description[' . $this->locale . ']'];
+        }
+        return $description;
+    }
+
+    public function getOgTitle()
+    {
+        $ogTitle = $this->getTitle();
+        if (isset($this->viewBagProperties['og_title'])) {
+            $ogTitle = $this->viewBagProperties['og_title'];
+        }
+
+        if (isset($this->viewBagProperties['og_title[' . $this->locale . ']'])) {
+            $ogTitle = $this->viewBagProperties['og_title[' . $this->locale . ']'];
+        }
+        return $ogTitle;
+    }
+
+    public function getOgDescription()
+    {
+        $ogDescription = $this->getDescription();
+        if (isset($this->viewBagProperties['og_description'])) {
+            $ogDescription = $this->viewBagProperties['og_description'];
+        }
+
+        if (isset($this->viewBagProperties['LocaleOg_description[' . $this->locale . ']'])) {
+            $ogDescription = $this->viewBagProperties['LocaleOg_description['. $this->locale . ']'];
+        }
+        return $ogDescription;
+    }
+
+    public function getOgImage()
+    {
+        $mediaUrl = url(Config::get('cms.storage.media.path'));
+        $ogImage = $this->getSiteImageFromSettings();
+        if ($settingsSiteImage = Settings::instance()->siteImage) {
+            $ogImage = $mediaUrl . $settingsSiteImage;
+        }
+
+        if (isset($this->viewBagProperties['og_image'])) {
+            $ogImage = $mediaUrl . $this->viewBagProperties['og_image'];
+        }
+        return $ogImage;
+    }
+
+    public function getOgVideo()
+    {
+        $ogVideo = null;
+        if (isset($this->viewBagProperties['og_video'])) {
+            $ogVideo = $this->viewBagProperties['og_video'];
+        }
+        return $ogVideo;
+    }
+
+    public function getOgType()
+    {
+        $ogType = $this->viewBagProperties['og_type'] ?? 'website';
+        return $ogType;
+    }
+
+    public function getSiteImageFromSettings()
+    {
+        $siteImage = null;
+        $mediaUrl = url(Config::get('cms.storage.media.path'));
```

Comment:
Why haven't you use `MediaLibrary`? AFAIR it has method to return the path to file directly.